ibc/store/bulk_store_cache.go

```diff
@@ -0,0 +1,188 @@
+package store
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/pokt-network/pocket/persistence/kvstore"
+	"github.com/pokt-network/pocket/runtime/configs"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/modules/base_modules"
+)
+
+var (
+	_ modules.BulkStoreCacher = &bulkStoreCache{}
+
+	cacheDirs = func(storesDir string) string { return fmt.Sprintf("%s/caches", storesDir) }
+)
+
+// bulkStoreCache holds an in-memory map of all the provable stores in use
+// RESEARCH: Look into parrelising the caching methods
+type bulkStoreCache struct {
+	base_modules.IntegrableModule
+
+	m sync.Mutex
```

Comment:
Add a comment on what this is for. If it's a mutext for the `stores` map, group them together inside the struct